service/src/test/java/org/databiosphere/workspacedataservice/pact/TDRPactTest.java

```diff
@@ -0,0 +1,125 @@
+package org.databiosphere.workspacedataservice.pact;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.PactSpecVersion;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.datarepo.model.SnapshotModel;
+import java.util.Map;
+import java.util.UUID;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoClientFactory;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoDao;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoException;
+import org.databiosphere.workspacedataservice.datarepo.HttpDataRepoClientFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+class TDRPactTest {
+
+  @BeforeEach
+  void setUp() {
+    // Without this setup, the HttpClient throws a "No thread-bound request found" error
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    // Set the mock request as the current request context
+    RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+  }
+
+  static final String dummySnapshotId = "12345678-abc9-012d-3456-e7fab89cd01e";
+
+  @Pact(consumer = "wds", provider = "datarepo")
+  public RequestResponsePact noSnapshotPact(PactDslWithProvider builder) {
+    return builder
+        .given("snapshot doesn't exist", Map.of("id", dummySnapshotId))
+        .uponReceiving("a snapshot request")
+        .path("/api/repository/v1/snapshots/" + dummySnapshotId)
+        .query("include=TABLES")
+        .method("GET")
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
+
+  @Pact(consumer = "wds", provider = "datarepo")
+  public RequestResponsePact noAccessToSnapshotPact(PactDslWithProvider builder) {
+    return builder
+        .given("user does not have access to snapshot")
```

Comment:
```suggestion
        .given("user does not have access to a snapshot with the given id", Map.of("id", dummySnapshotId))
```

Piggybacking on @okotsopoulos ' suggestion, I'd recommend providing the snapshot ID in each given, like you do on line 47, so the provider can stub/mock out specifically for the behavior that should be simulated for that ID.

I started using a convention "with the given *id*" as a readable call to action for the provider implementor to look in the params hash for a variable named *id*.